code-challenges.test.js

```diff
@@ -14,39 +14,167 @@
 
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("secretCode", () => {
+    it("should change 'lackadaisical' to 'L4ck4d41s1c4l'", () => {
+        const secretCodeWord1 = "Lackadaisical"
+        expect(secretCode(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+    })
+    // Expected output: "L4ck4d41s1c4l"
+    
+    it("should change 'Gobbledygook' to 'G0bbl3dyg00k'", () => {
+        const secretCodeWord2 = "Gobbledygook"
+        expect(secretCode(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+    })
+    // Expected output: "G0bbl3dyg00k"
+    it("should change 'Eccentric' to '3cc3ntr1c'", () => {
+        const secretCodeWord3 = "Eccentric"
+        expect(secretCode(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+
+    // Expected output: "3cc3ntr1c"
+    
+    //Test Suites: 1 failed, 1 total
+    //Tests:       0 total
+    //Snapshots:   0 total
+    //Time:        0.521 s
+
 
 // b) Create the function that makes the test pass.
 
+// pseudocode 
+// declare a funtion codedword
+//itirate through the array and find the strings that match "a, e, i, o,u"
+// make an object to make a table to map out the characters to their corresponding values
+// the coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
+// we use the replace method 
+// use .toLowercase to catch letters that are capitilized
+// we use charCodeMap[char.toLowercase()] to retrieve the corresponding matched character from the object
+// we need to return the orinal character for the characters that arent in our map
+// well use the OR '||' operator for the last step 
+// the output will be our encoded message 
+
```

Comment:
Code looks great! However, your pseudocode suggests you are not fully understanding what is happening here. You are taking parts of the code and putting them into your pseudocode, when it should be the other way around. For example, yes you are using `charCodeMap[char.toLowerCase()]`, but that's not what is retrieving the values at the key that matches the search taking place through the `.replace()` method using regex. Make sure you break the different parts down and are able to fully explain what is going on.